meta-balena-common/recipes-containers/hostapp-update/files/hostapp-update

```diff
@@ -156,5 +187,15 @@ mv "$SYSROOT/counter.new" "$SYSROOT/counter"
 sync -f "$SYSROOT"
 
 if [ "$reboot" = 1 ]; then
-	reboot
+  # do a supervisor reboot so that we check for update locks after HUP,
+  # failure to reboot will be a failure in the HUP procedure
+ 
+  # check if supervisor is running first
+  if ! call_supervisor_api "$PING_ENDPOINT"; then
+    exit 1 
+  fi
+
+  if ! call_supervisor_api "$REBOOT_ENDPOINT"; then
+    exit 1
+  fi
```

Comment:
I would add a final log message that makes it clear that the HUP is waiting on the supervisor to reboot the device.